### PR TITLE
General: Move mongo db logic and remove avalon repository

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Brief description
Moved last piece of used code from avalon repository into openpype and removed the repository.

## Description
Moved schema validation and mongo connection logic with current context `Session` object logic to OpenPype. Some code is modified to reuse already existing code in OpenPype and some deprecated code is removed completely. Changed imports to use moved code. Submodule `avalon-core` is removed from openpype repository.

## Additional info
Moved `avalon.mongodb` and `AvalonMongoDB`. Objects of `AvalonMongoDB` are now using openpype's mongo connection instead of creating new, removed `AvalonMongoConnection`. Removed handling of unused environment variables related to avalon: `AVALON_MONGO`, `AVALON_CONFIG`, `AVALON_SCHEMA`, `AVALON_DEADLINE`, `AVALON_SILO`, `AVALON_DEBUG`, `AVALON_UPLOAD`, `AVALON_USERNAME`, `AVALON_PASSWORD`, `AVALON_LOCATION` and `AVALON_SENTRY`.

Content of `avalon.io` was moved to `openpype.pipeline.legacy_io` with global `Session` hoding current context.

## Testing notes:
- all hosts should be successfully installed:
    - [x] Maya
    - [x] Flame
    - [x] Houdini
    - [x] Nuke
    - [x] Hiero
    - [ ] Fusion
    - [ ] Resolve
    - [x] AfterEffects
    - [x] Photoshop
    - [x] Harmony
    - [x] TVPaint
    - [ ] Unreal
    - [x] Blender
    - [x] Standalone publisher
    - [x] Webpublisher

- all tools should work as expected:
    - [x] Launcher
    - [x] Project manager
    - [x] Settings
    - [x] Select context dialog
    - [x] Standalone publisher
    - [x] Tray publisher
    - [x] Creator
    - [x] Workfiles
    - [x] Loader
    - [x] Library loader
    - [x] Scene inventory
    - [ ] Texture copy
    - [x] Maya look assigner
    - [x] Adobe webserver

### CODE REVIEW

**Review by mkolar — COMMENTED** [review]
It is still expecting `repos` folder, but other than that I've tested in Maya, Houdini and Blender and all tools are good

**Comment by iLLiCiTiT** [comment]
> It is still expecting repos folder

Changed

**Comment by kalisp** [comment]
Build fails on empty environment with
[build.log](https://github.com/pypeclub/OpenPype/files/8520524/build.log)

File: setup.py [hunk 1]
```diff
@@ -123,12 +123,15 @@ def validate_thirdparty_binaries():
...
-    "repos",
     "schema",
     "LICENSE",
     "README.md"
 ]
 
+repos_path = openpype_root / "repos"
+if repos_path.exists():
+    include_files.append("repos")
+
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply setup.py:134]: We've discussed it and we don't want to remove it yet as we may need to add other repositories in near future.

[the remark this answers is not in the document]
Comment (BigRoy) [reply setup.py:134]: Shouldn't we still remove it now and when we know "repos" should be there then explicitly add it back in as a required one? I'm a bit worried about this potentially failing silently unexpectedly.

[the remark this answers is not in the document]
Comment (mkolar) [reply setup.py:134]: You're probably right and we could just add the folder back when needed.

[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply setup.py:134]: Created new PR into this PR https://github.com/pypeclub/OpenPype/pull/3087 .

**Comment by jakubjezek001** [comment]
Flame works as expected..

**Pull request merged**